backend/tessend/data.go

```diff
@@ -0,0 +1,100 @@
+package tessend
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	v2 "github.com/sensu/sensu-go/api/core/v2"
+)
+
+const (
+	// TimestampFormat is the string representation for the license timestamps e.g.
+	// "2018-07-26T12:12:06-04:00"
+	TimestampFormat = time.RFC3339
+)
+
+// Data is the payload sent to tessen
+type Data struct {
+	// General information about the Sensu installation.
+	Cluster Cluster `json:"cluster"`
+
+	// Metric data.
+	Metrics v2.Metrics `json:"metrics"`
+}
+
+// Cluster is the cluster information sent to tessen
+type Cluster struct {
+	// ID is the ID of the sensu-enterprise-go cluster.
+	ID string `json:"id"`
+
+	// Version is the Sensu release version (e.g. "1.4.1").
+	Version string `json:"sensu_version"`
+
+	// License is the Cluster's license.
+	License License
+}
+
+// Timestamp is an alias to time.Time with json Marshaling/Unmarshaling support
+type Timestamp time.Time
+
+// MarshalJSON implements the json.Marshaler interface.
+func (t Timestamp) MarshalJSON() ([]byte, error) {
+	stamp := fmt.Sprintf("%q", time.Time(t).Format(TimestampFormat))
+	return []byte(stamp), nil
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (t *Timestamp) UnmarshalJSON(b []byte) error {
+	var str string
+	if err := json.Unmarshal(b, &str); err != nil {
+		return fmt.Errorf("Cannot unmarshal the license timestamp")
+	}
+
+	value, err := time.Parse(TimestampFormat, str)
+	if err != nil {
+		return err
+	}
+
+	*t = Timestamp(value)
+	return nil
+}
+
+// License holds information about a user's enterprise software license,
+// including duration of validity and enabled features.
+type License struct {
```

Comment:
We don't actually need to define it. Since we don't manipulate it, we could use an `interface{}` to hold it so at line 35 above, we can simply do `License interface{}`.

Reply:
I actually tried this, but because there's custom marshal/unmarshal functions for `Timestamp`, Tessen was rejecting the data.

Reply:
Open to other ways to work around defining the license data here!